go/libraries/doltcore/doltdb/root_val.go

```diff
@@ -135,6 +137,234 @@ func (root *RootValue) HasTable(ctx context.Context, tName string) (bool, error)
+
+	if err != nil {
+		return nil, false, err
+	}
+
+	if !found && !tblFound {
+		// table doesn't exist in current commit or in history
+		return nil, false, nil
+	}
+
+	if tblFound {
+		sch, err := t.GetSchema(ctx)
+
+		if err != nil {
+			return nil, false, err
+		}
+
+		err = ss.AddSchemas(sch)
+
+		if err != nil {
+			return nil, false, err
+		}
+	}
+
+	return ss, true, err
+}
+
+// GetUniqueTagFromNomsKinds returns a tag that has not yet been used in the history of this root.
+func (root *RootValue) GetUniqueTagFromNomsKinds(ctx context.Context, schKinds []types.NomsKind) (uint64, error) {
+	ssMap, err := root.getOrCreateSuperSchemaMap(ctx)
+
+	if err != nil {
+		return schema.InvalidTag, err
+	}
+
+	var sss []*schema.SuperSchema
+	err = ssMap.Iter(ctx, func(key, value types.Value) (stop bool, err error) {
+		ssValRef := value.(types.Ref)
+		ssVal, err := ssValRef.TargetValue(ctx, root.vrw)
+
+		if err != nil {
+			return true, err
+		}
+
+		ss, err := encoding.UnmarshalSuperSchemaNomsValue(ctx, root.vrw.Format(), ssVal)
+
+		if err != nil {
+			return true, err
+		}
+
+		sss = append(sss, ss) // go get -f parseltongue
+		return false, nil
+	})
+
+	if err != nil {
+		return schema.InvalidTag, err
+	}
+
+	rootSuperSchema, err := schema.SuperSchemaUnion(sss...)
+
+	if err != nil {
+		return schema.InvalidTag, err
+	}
+
+	// super schemas are only persisted on commit, so add in working schemas
+	tblMap, err := root.getTableMap()
+
+	if err != nil {
+		return schema.InvalidTag, err
+	}
+
+	err = tblMap.Iter(ctx, func(key, _ types.Value) (stop bool, err error) {
+		tbl, _, err := root.GetTable(ctx, string(key.(types.String)))
+		if err != nil {
+			return true, err
+		}
+		sch, err := tbl.GetSchema(ctx)
+		if err != nil {
+			return true, err
+		}
+		err = rootSuperSchema.AddSchemas(sch)
+		if err != nil {
+			return true, err
+		}
+		return false, nil
+	})
+
+	return schema.AutoGenerateTag(rootSuperSchema, schKinds), nil
+}
+
+// GerSuperSchemaMap returns the Noms map that tracks SuperSchemas, used to create new RootValues on checkout branch.
+func (root *RootValue) GetSuperSchemaMap(ctx context.Context) (types.Map, error) {
+	return root.getOrCreateSuperSchemaMap(ctx)
+}
+
+func (root *RootValue) getStaleSuperSchema(ctx context.Context, tName string) (*schema.SuperSchema, bool, error) {
+	ssm, err := root.getOrCreateSuperSchemaMap(ctx)
+
+	if err != nil {
+		return nil, false, err
+	}
+
+	v, found, err := ssm.MaybeGet(ctx, types.String(tName))
+
+	if err != nil {
+		return nil, false, err
+	}
+	if !found {
+		// Super Schema doesn't exist for new or nonexistent table
+		return nil, false, nil
+	}
+
+	ssValRef := v.(types.Ref)
+	ssVal, err := ssValRef.TargetValue(ctx, root.vrw)
+
+	if err != nil {
+		return nil, false, err
+	}
+
+	ss, err := encoding.UnmarshalSuperSchemaNomsValue(ctx, root.vrw.Format(), ssVal)
+
+	if err != nil {
+		return nil, false, err
+	}
+
+	return ss, true, nil
+}
+
+// TODO: get or create from history of root
```

Comment:
it's an outdated comment. 
When I was writing SuperSchema I wanted to be sure that new SuperSchemas written to legacy repos would include the entire history of the table. 
Because SuperSchemas are being landed at the same same time as unique tags, we don't have to worry about that here. SuperSchemas will be generated as part of the migration rebase.